### DISCUSSION TITLE
Upgrading RainbowKit to support wagmi v2 - Connector Changes

### DISCUSSION THREAD

**Discussion by DannyDelott** [rainbow-me/rainbowkit#1539] [General]
Hello everyone!

We're looking to upgrade RainbowKit to support the latest version of wagmi, specifically v2. The new version introduces some changes that affect our current implementation.

For reference, here's the discussion on wagmi v2: [Wagmi v2 Discussion](https://github.com/wagmi-dev/wagmi/discussions/3068).

One of the main challenges is the change in the shape of `connectors` returned from `getDefaultWallets`. Here's a snippet of our current implementation:

```ts
import { createConfig, http } from "wagmi";
import { mainnet } from "wagmi/chains";
import { getDefaultWallets } from "@rainbow-me/rainbowkit";

const { connectors } = getDefaultWallets({
  appName: "my app",
  projectId: 'project-id',
  chains: [mainnet],
});

export const wagmiConfig = createConfig({
  chains: [mainnet],
  connectors, // <-- THIS is where the incompatibility lies
  transports: {
    [mainnet.id]: http('alchemy-url'),
  },
});
```

Would love to hear your thoughts, suggestions, or any potential solutions to address this.

**Comment by art-eth** [comment]
Looks like wagmi/viem makes it possible to change chains while being disconnected now so it could be a good additon to RainbowKit as well 

+1

**Comment by DanielSinclair** [comment]
@DannyDelott Thanks for spotting this. We are working with the Wagmi team on starting this migration. It will mean breaking changes for RainbowKit, so you probably won't see a production release for another ~1.5 months. The Wagmi release is expected to remain in alpha during that time.

**Reply by art-eth** [reply]
Great news! Looking forward to RainbowKit 2.0.0 alpha :)

I am about to rewrite my website to support multiple chains etc, so I can test a lot

**Reply by songkeys** [reply]
Is it mature to release alphas for this PR now? https://github.com/rainbow-me/rainbowkit/pull/1601

**Comment by tuler** [comment]
Now that wagmi v2 is released, is there a public beta of rainbowkit that supports it?

**Comment by DmyKov** [comment]
When can we expect for migration?

**Comment by JohnnyGoodpick** [comment]
Also interested to know if there's an estimated release date for this

**Comment by DanielSinclair** [comment]
We'll release the RainbowKit beta with Wagmi v2 support over the weekend, and I'll make sure to update the thread here.

**Reply by launchthatbrand** [reply]
Is this out somewhere?

**Comment by aiiiden** [comment]
Thank you for supporting efforts to enable wagmi v2 for the rainbow-kit.
Apparently, there will be no more breaking changes after v2.

> Wagmi v2 should be the last major version that will have this many actionable breaking changes.
> https://wagmi.sh/react/guides/migrate-from-v1-to-v2

**Comment by DanielSinclair** [comment]
The `beta` release for RainbowKit v2 with Wagmi v2 support is now available. Here is a [beta migration guide.](https://beta.rainbowkit.com/guides/rainbowkit-wagmi-v2) Please let us know if you see any type issues, bugs, or have feedback! We'll be in beta for the next week or so.

**Reply by endresol** [reply]
The migration guide is not very complete.
The RainbowKitProvider import is missing and reusing the name config, that first came from the wagmi repo and now from rainbowkit, is a bit confusing.

**Reply by jmrossy** [reply]
@DanielSinclair So far everything's work for me, thanks

**Reply by endresol** [reply]
It works, just a comment to make it better.

**Comment by arlgnv** [comment]
@DanielSinclair Hi. Is there some news about when rainbow-kit with wagmi v.2 support will be out in prod?

**Reply by SwarmMonkey** [reply]
I also need this. It is stopping me from using rainbowkit and I actually want to.

**Reply by c0rdeiro** [reply]
@PatrickManfra rainbowkit supports this since v2

**Reply by DanielSinclair** [reply]
Correct, RainbowKit 2.0 and above supports Wagmi v2
